new/detector/implementation/custom/filter.go

```diff
@@ -45,7 +45,7 @@ func matchFilter(
 
 	if filter.Detection != "" {
 		effectiveScope := filter.Scope
-		if scope == settings.RESULT_SCOPE {
+		if filter.Scope == settings.NESTED_SCOPE && scope == settings.RESULT_SCOPE {
```

Comment:
Not sure if this is clearer 🤔 

```suggestion
		if effectiveScope == settings.NESTED_SCOPE && scope == settings.RESULT_SCOPE {
```